proposals/proposal-template.md

```diff
@@ -0,0 +1,98 @@
+# JSON Schema Proposal: <TODO>
+
+## Abstract
+
+<!--
+Fill in the specification(s) that will change.  If adding a keyword, which vocabulary
+will contain it?
+
+For example
+
+    This document proposes a change to the JSON Schema Core specification and
+    Applicator vocabulary by adding the `propertyDependencies` keyword.
+
+-->
+
+This document proposes a change to the ??? specification by adding ???.
+
+## Status
+
+**Current Status**: PROPOSAL
+
+<!-- TODO: We should have a short standard blurb outlining the stages involved in a
+feature making its way to stable status. -->
+
+<!-- TODO: Link to a document that describes the proposal => stable process in
+detail. -->
```

Comment:
Starting to think this can go away completely.  The document _is_ a proposal.

Maybe we want to keep it around for posterity, but I think we could use file paths to handle that.  Maybe something like moving the files to a `completed/<release>/` folder and renaming the file to `<filename>-rejected.md` / `<filename>-accepted.md`.

Personally, I'm happy with just deleting the file once we don't need it anymore, whether it was rejected or accepted.

Reply:
I think it would be a good idea to use the status field to communicate the maturity of the proposal. For example, something like,

1. Development -- The proposal is in the early stages of being considered
2. Release Candidate -- No further changes are expected, but we want to see as much implementation and usage to flush out any remaining issues
3. Awaiting Release -- The proposal has been excepted and will be included in the next release. There will be no more changes.
4. Released
5. Rejected

We don't even need to have a fixed list of statuses we're limited to. I think what's important is that it communicate something useful.

As for whether to keep around completed proposals, I think it's best to keep them around. Technically, they're never gone because they're in the git history, but people aren't going to go digging through our git history to find a proposal they're watching. I'm imagining someone trying to find the status of a proposal they're interested in and being confused because they can't find it or any explanation why it disappeared. For the same reason, I think it's best not to move them to a "rejected" or "completed" folder because that would break any bookmarks people have to those proposals. They can maybe be removed after some generous period of time, but not right away.

Reply:
I'm happy keeping this as an open text field.  I'll update so that's clear.